src/SoCreate.ServiceFabric.PubSub/State/ReferenceWrapper.cs

```diff
@@ -92,13 +99,15 @@ public bool ShouldDeliverMessage(MessageWrapper message)
         {
             if (!(MessageWrapperExtensions.PayloadSerializer is DefaultPayloadSerializer))
                 return true;
-            if (_routingKeyValue == null)
+            if (RoutingKeyName is null || RoutingKeyValue is null)
                 return true;
 
             var token = MessageWrapperExtensions.PayloadSerializer.Deserialize<JToken>(message.Payload);
-            string value = (string)token.SelectToken(_routingKeyValue[0]);
-
-            return string.Equals(_routingKeyValue[1], value, StringComparison.InvariantCultureIgnoreCase);
+            string value = (string)token.SelectToken(RoutingKeyName);
+            if(value is null)
+                return false;
+            var regex = new Regex("^" + Regex.Escape(RoutingKeyValue).Replace(@"\*", ".*") + "$");
```

Comment:
This makes me think we might as well just go all the way and support any regex for the RoutingKey instead of inventing a custom and less flexible wildcard scheme.

Reply:
That is good idea but it would be a breaking change that would only show up while running which sounds dangerous.

Reply:
What are your thoughts on adding a new bool to the Subscribe Attribute, something like ```UseRegexForRouting```.

```
if (UseRegexForRouting) {
// regex compare
} else {
// string.equals
}```

This wouldn't prevent any breaking changes, if someone had used regex reserved characters in there RoutingKey.